main/lxc/lxc.initd

```diff
@@ -51,15 +51,36 @@ checkconfig() {
 	fi
 }
 
+systemd_ctr() {
+	# required for lxc-console & services inside systemd containers
+	local cgroup=/sys/fs/cgroup/systemd cmd=$1
+	local mnt_opts='rw,nosuid,nodev,noexec,relatime,none,name=systemd'
+
+	case "$cmd" in
+		 mount) checkpath -d $cgroup
+			if ! mount | grep $cgroup 1>/dev/null; then
+				mount -t cgroup -o $mnt_opts cgroup $cgroup
+			fi
+			;;
+	       unmount) if mount | grep $cgroup 1>/dev/null; then
+				umount $cgroup
+			fi
+			;;
+	esac
+}
+
 _autostart() {
 	ebegin "$1 LXC containers"
 	shift
-	lxc-autostart --group "${LXC_GROUP:-onboot,}" "$@"
+	lxc-autostart --group "${lxc_group:-onboot,}" "$@"
 	eend $?
 }
 
 start() {
 	checkconfig || return 1
+	case "$systemd_container" in
+		y*|Y*) systemd_ctr mount;;
+	esac
```

Comment:
I prefer using openrc's `yesno` for this. That way things will work even with things like `systemd_container=1`.

````sh
if yesno "$systemd_container"; then
    systemd_ctr mount
fi

Reply:
`yesno` returns zero for `Y` or `y` so `confd` changed to `ON`

adding the descriptions prevents `rc-service lxc describe` putting warnings in the logs:
```
Nov 16 23:06:00 xxxx daemon.warn /etc/init.d/lxc[28565]: No description for lxc
Nov 16 23:06:00 xxxx daemon.warn /etc/init.d/lxc[28566]: ^[[1m^[[36mreboot^[[m: no description
```

Reply:
> `yesno` returns zero for `Y` or `y` so `confd` changed to `ON`

Yes, it returns 0 (aka true) for `Y`, `y`, `yes` and similar, 1 (aka false) for anything else. I really don’t understand why you used `ON` because of this.

Reply:
it [does not return true](http://linuxhowtos.org/manpages/8/openrc-run.htm) for `Y` or `y`:
```
yesno value
    If value matches YES, TRUE, ON or 1 regardless of case then we return 0, otherwise 1. 
```
I had the config file as `Y` before & it didn't work - which is why I set it to `ON`